kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/ProcessServiceIntegrationTest.java

```diff
@@ -428,6 +428,79 @@ public void testManipulateProcessVariable() throws Exception {
         }
 
     }
+    
+    @Test
+    public void testProcessVariablesDataTypes() throws Exception {
+        Long processInstanceId = processClient.startProcess(CONTAINER_ID, "variables.types");
+        assertNotNull(processInstanceId);
+        assertTrue(processInstanceId>0);
+        try {
+            
+            short age = 44;
+            processClient.setProcessVariable(CONTAINER_ID, processInstanceId, "age", age );
+            assertEquals( age, ((Short)processClient.getProcessInstanceVariable(CONTAINER_ID, processInstanceId, "age")).shortValue());
+            
+            boolean married = true;
+            processClient.setProcessVariable(CONTAINER_ID, processInstanceId, "married", married );
+            assertEquals( married, ((Boolean)processClient.getProcessInstanceVariable(CONTAINER_ID, processInstanceId, "married")).booleanValue());
+            
+            String name = "Javierito";
+            processClient.setProcessVariable(CONTAINER_ID, processInstanceId, "name", name );
+            assertEquals( name, (String)processClient.getProcessInstanceVariable(CONTAINER_ID, processInstanceId, "percentage"));
+            
+            float percentage = 99.9f;
+            processClient.setProcessVariable(CONTAINER_ID, processInstanceId, "percentage", percentage);
+            assertEquals( percentage, ((Float)processClient.getProcessInstanceVariable(CONTAINER_ID, processInstanceId, "percentage")).floatValue(), 0.001f);
```

Comment:
```suggestion
            assertEquals( percentage, (Float)processClient.getProcessInstanceVariable(CONTAINER_ID, processInstanceId, "percentage"), 0.001f);
```

Reply:
Here floatValue is needed to execute assertEquals (float, float)